src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1051,18 +1103,8 @@ def apply_fixes(self, dialect, rule_code, fixes):
                                         f.edit_type, f
                                     )
                                 )
-                            # We've applied a fix here. Move on, this also consumes the
-                            # fix
-                            # TODO: Maybe deal with overlapping fixes later.
-                            break
-                        else:
-                            # We've not used the fix so we should keep it in the list
-                            # for later.
-                            unused_fixes.append(f)
                     else:
                         seg_buffer.append(seg)
-                # Switch over the the unused list
-                fixes = unused_fixes + fix_buff
```

Comment:
Note how this tricky "end of loop" logic all goes away now that we're using a dictionary instead.